service/interfaces.go

```diff
@@ -5,21 +5,24 @@ import (
 	"net/http"
 
 	"github.com/ONSdigital/dp-api-clients-go/v2/cantabular"
+	"github.com/ONSdigital/dp-api-clients-go/v2/dataset"
 	"github.com/ONSdigital/dp-healthcheck/healthcheck"
 	"github.com/ONSdigital/dp-population-types-api/config"
 )
 
 //go:generate moq -out mock/initialiser.go -pkg mock . Initialiser
 //go:generate moq -out mock/server.go -pkg mock . HTTPServer
-//go:generate moq -out mock/healthCheck.go -pkg mock . HealthChecker
-//go:generate moq -out mock/cantabularClient.go -pkg mock . CantabularClient
+//go:generate moq -out mock/health_check.go -pkg mock . HealthChecker
+//go:generate moq -out mock/cantabular_client.go -pkg mock . CantabularClient
+//go:generate moq -out mock/dataset_api_client.go -pkg mock . DatasetAPIClient
 
 // Initialiser defines the methods to initialise external services
 type Initialiser interface {
 	GetHTTPServer(bindAddr string, router http.Handler) HTTPServer
 	GetCantabularClient(cfg config.CantabularConfig) CantabularClient
 	GetHealthCheck(cfg *config.Config, time, commit, version string) (HealthChecker, error)
 	GetResponder() Responder
+	GetDatasetAPIClient(cfg *config.Config) DatasetAPIClient
```

Comment:
I have put in a lot of this infrastructure work also, have we repeated work? looks like a lot of work has been duplicated?

Reply:
Not a problem, will resolve when rebasing